core/src/main/java/com/coderanch/blackjack/Card.java

```diff
@@ -12,31 +12,66 @@
  */
 final class Card {
   
+  enum Rank { ACE, TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN, JACK, QUEEN, KING };
+  enum Suit { SPADES, HEARTS, CLUBS, DIAMONDS };
+  
+  private final Rank rank;
+  private final Suit suit;
+  
   /**
-   * The rank of a {@link Card}.
-   */
-  enum Rank {}
-
-  /**
-   * The suit of a {@link Card}.
+  * Constructor.
+  *
+  * @param rank (required) Rank value. Must have value of enum Rank. 
+  * @param suit (required) Suit value. Must have value of enum Suit.
+  */
+  
+  public Card(Rank rank, Suit suit) {
+    this.rank = getConstRank(rank);
+    this.suit = getConstSuit(suit);
+  }
+  
+  /*
+  * Oracle states that private methods should not have a JavaDoc, 
+  * also that assertions should not be used for runtime exceptions.
+  * I have implemented these getters as I;ve been told logic should not be placed
+  * in the constructor itself. I'm not sure if it is okay to do it this way? 
+  * Any feedback would be geat. Also, suit not being of Suit is an error (cannot find symbol).
+  * How is this supposed to be checked for without surrounding constructor in try catch?
```

Comment:
Instead of calling these methods to return a default value, throw an exception when a parameter is invalid. One of the unit tests actually tests this, so you will have written the correct solution when the tests pass.

How do you compile the code? Do you have Maven installed on your system? Are you using an IDE?

Reply:
@Nibsi I normally write code in notepad++, and then I use the command line to javac and test with either another class or I add code to it for output. I have Eclipse and NetBeans, but I've been practicing for the OCA, so I've been trying to do it the hard way to learn more. Honestly, I didn't realize how bad I was at writing code until I started doing it this way. Although, this last time I edited the class here on the GitHub website.

Reply:
@smcdonald4812 It's fine to use Notepad++, but since this is a Maven project, I suggest you download and install Maven and then compile the code using `mvn package`. That way the tests are run and you can see what changes you have to make before you push a broken build.